test/engine/update.result

```diff
@@ -850,3 +850,305 @@ s:update({1}, {{'=', 'field3', {d = 30, e = 31, f = 32}}})
 s:drop()
 ---
 ...
+--
+-- gh-3378: allow update absent nullable fields
+-- '!'
+s = box.schema.create_space('test')
+---
+...
+pk = s:create_index('pk')
+---
+...
+s:replace{1, 2}
+---
+- [1, 2]
+...
+s:update({1}, {{'!', 4, 0}})
+---
+- [1, 2, null, 0]
+...
+s:drop()
+---
+...
+-- '#'
+s = box.schema.create_space('test')
+---
+...
+pk = s:create_index('pk')
+---
+...
+s:replace{1, 2}
+---
+- [1, 2]
+...
+s:update({1}, {{'#', 4, 1}})
+---
+- [1, 2]
+...
+s:drop()
+---
+...
+-- Update respects field_count
+s = box.schema.create_space('test', {field_count = 2})
+---
+...
+pk = s:create_index('pk')
+---
+...
+s:replace{1, 2}
+---
+- [1, 2]
+...
+s:update({1}, {{'!', 3, 0}})
+---
+- error: Tuple field count 3 does not match space field count 2
+...
+s:update({1}, {{'=', 3, 0}})
+---
+- error: Tuple field count 3 does not match space field count 2
+...
+s:drop()
+---
+...
+-- '='
+s = box.schema.create_space('test')
+---
+...
+pk = s:create_index('pk')
+---
+...
+s:replace{1, 2}
+---
+- [1, 2]
+...
+s:update({1}, {{'=', 4, 0}})
+---
+- [1, 2, null, 0]
+...
+s:drop()
+---
+...
+-- Ops that must still fail
+format = {}
+---
+...
+format[1] = {name = 'field1', type = 'unsigned'}
+---
+...
+format[2] = {name = 'field2', type = 'unsigned'}
+---
+...
+format[3] = {name = 'field3', type = 'unsigned', is_nullable = true}
+---
+...
+s = box.schema.create_space('test', {format = format})
+---
+...
+pk = s:create_index('pk')
+---
+...
+s:replace{1, 2, box.NULL}
+---
+- [1, 2, null]
+...
+s:update({1}, {{'+', 3, 0}})
```

Comment:
Didn't get the purpose of the test: I believe the fact that update operation involving null and actual value is checked somewhere else.